sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusClientFactory.cs

```diff
@@ -0,0 +1,226 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using Azure.Core;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
+using Microsoft.Azure.WebJobs.Extensions.Clients.Shared;
+using Microsoft.Azure.WebJobs.ServiceBus.Listeners;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.ServiceBus
+{
+    /// <summary>
+    /// This class provides factory methods for the creation of instances
+    /// used for ServiceBus message processing.
+    /// </summary>
+    internal class ServiceBusClientFactory
+    {
+        private readonly ServiceBusOptions _options;
+        private readonly IConfiguration _configuration;
+        private readonly AzureComponentFactory _componentFactory;
+
+        private readonly ConcurrentDictionary<string, ServiceBusSender> _messageSenderCache = new();
+        private readonly ConcurrentDictionary<string, ServiceBusReceiver> _messageReceiverCache = new();
+        private readonly ConcurrentDictionary<string, ServiceBusClient> _clientCache = new();
+        private readonly ConcurrentDictionary<string, ServiceBusProcessor> _processorCache = new();
```

Comment:
Can we cache processors? What if the customer has multiple functions using use same connection?

Reply:
As long as it is looking at the same queue/subscription it can be used across functions.

Reply:
I thought the processor only allows a single Process event subscription.

Reply:
Yeah, you are right. Listeners are 1:1 with functions - I was thinking that a listener could be used across functions for some reason. Will update.